lib/index.js

```diff
@@ -1,98 +1,152 @@
-import { find, filter } from 'lodash/collection'
-import createClients from 'contentful-batch-libs/dist/utils/create-clients'
-import pushToSpace from 'contentful-batch-libs/dist/push/push-to-space'
 import Promise from 'bluebird'
-import transformSpace from 'contentful-batch-libs/dist/transform/transform-space'
-import getTransformedDestinationResponse from './get-transformed-destination-response'
-import log from 'npmlog'
-import { startCase } from 'lodash'
 import Table from 'cli-table2'
+import Listr from 'listr'
+import UpdateRenderer from 'listr-update-renderer'
+import VerboseRenderer from 'listr-verbose-renderer'
+import { startCase } from 'lodash'
 import moment from 'moment'
 
+import pushToSpace from 'contentful-batch-libs/dist/push/push-to-space'
+import transformSpace from 'contentful-batch-libs/dist/transform/transform-space'
+import createClients from 'contentful-batch-libs/dist/utils/create-clients'
+import {
+  setupErrorLogging,
+  displayErrorLog,
+  writeErrorLogFile
+} from 'contentful-batch-libs/dist/utils/logging'
+
+import getTransformedDestinationResponse from './get-transformed-destination-response'
 import parseOptions from './parseOptions'
 
-const summary = {}
+function createListrOptions (options) {
+  if (options.useVerboseRenderer) {
+    return {
+      renderer: VerboseRenderer
+    }
+  }
+  return {
+    renderer: UpdateRenderer,
+    collapse: false
+  }
+}
 
 export default function runContentfulImport (params) {
-  summary.startTime = moment()
+  const summary = {
+    startTime: moment()
+  }
+  const errorLog = []
+  const timestamp = summary.startTime.format('YYYY-MM-DDTHH-mm-SS')
 
   const options = parseOptions(params)
+  const errorLogFilePath = `${process.cwd()}/contentful-import-error-log-${options.spaceId}-${timestamp}.json`
```

Comment:
we have `--error-log-file`, we should check if it is defined otherwise we define our own, maybe move all of this to `parseOptions` module